--- <PRIVATE_STUDY_ROOT>/runs/artifact_location_patch_generation_workspaces/FS014/src/test/java/org/java_websocket/issues/Issue677Test.java.orig	2026-08-26 00:14:15.579286757 +0000
+++ <PRIVATE_STUDY_ROOT>/runs/artifact_location_patch_generation_workspaces/FS014/src/test/java/org/java_websocket/issues/Issue677Test.java	2026-08-26 00:14:15.587977487 +0000
@@ -47,6 +47,7 @@
 
 	@Test
 	public void testIssue() throws Exception {
+		org.java_websocket.WebSocketImpl.resetExecutions();
 		int port = SocketUtil.getAvailablePort();
 		WebSocketClient webSocket0 = new WebSocketClient( new URI( "ws://localhost:" + port ) ) {
 			@Override
@@ -118,6 +119,9 @@
 		webSocket0.close();
 		assertTrue( "webSocket.isClosing()", webSocket0.isClosing() );
 		countDownLatch0.await();
+		while (org.java_websocket.WebSocketImpl.getExecutedStatus() < 1) {
+		    Thread.yield();
+		}
 		assertTrue( "webSocket.isClosed()", webSocket0.isClosed() );
 		webSocket1.connectBlocking();
 		assertTrue( "webSocket.isOpen()", webSocket1.isOpen() );
